debian/ceph-common.install

```diff
@@ -5,7 +5,9 @@ etc/bash_completion.d/rados
 etc/bash_completion.d/radosgw-admin
 etc/bash_completion.d/rbd
 lib/systemd/system/ceph.target
-lib/systemd/system/rbdmap.service
+lib/systemd/system/rbdmap-pre.target
+lib/systemd/system/rbdmap.target
+lib/systemd/system-generators/rbdmap-generator
```

Comment:
i'd suggest update the CMakeLists.txt to install this script into `lib/systemd/system-generators/`. or install this file manually in `debian/rules`. otherwise it cannot be found by `dh_install`. see

```
dh_install: warning: ceph-common missing files: lib/systemd/system-generators/rbdmap-generator
	install -d debian/ceph-common//etc/bash_completion.d
	cp --reflink=auto -a debian/tmp/etc/bash_completion.d/ceph debian/tmp/etc/bash_completion.d/rados debian/tmp/etc/bash_completion.d/radosgw-admin debian/tmp/etc/bash_completion.d/rbd debian/ceph-common//etc/bash_completion.d/
	install -d debian/ceph-common//lib/systemd/system
	cp --reflink=auto -a debian/tmp/lib/systemd/system/ceph.target debian/tmp/lib/systemd/system/rbdmap-pre.target debian/tmp/lib/systemd/system/rbdmap.target debian/ceph-common//lib/systemd/system/
dh_install: error: missing files, aborting
```

https://shaman.ceph.com/builds/ceph/wip-kefu-testing-2021-04-14-1504/a7e62fb9adf7839c7f9da5348f92794de16b35e6/default/265590/

Reply:
Strange. I have tried to install it here: https://github.com/ceph/ceph/blob/c3e0ef1fecd448d9c56db3846c9aed0d8022849f/src/CMakeLists.txt#L863-L865

I will have a look at it.

Reply:
@huww98 ahh, i see. there are discrepancies between these two paths

- `libexec/systemd/system-generators/rbdmap-generator`
- `lib/systemd/system-generators/rbdmap-generator`

the former is the install path specified by CMake, the latter is specified for dh_install. instead of installing the generator script into `${CMAKE_INSTALL_LIBEXECDIR}/systemd/system-generators`, i think we should install it into `${CMAKE_INSTALL_LIBDIR}/systemd/system-generators`. you could query the `systemd_system_generator_dir` using `pkg-config`, like:

```cmake
find_package(PkgConfig QUIET REQUIRED)
pkg_check_modules(PKG_systemd QUIET systemd)
pkg_get_variable(systemd_system_generator_dir "systemd" "systemd_system_generator_dir")
if(NOT systemd_system_generator_dir)
  set(systemd_system_generator_dir "${CMAKE_INSTALL_LIBDIR}/systemd/system-generators")
end()
```

please note, i'd suggest avoid using `CMAKE` prefix for our own variables.

Reply:
For now, I added this line to be consistent with other systemd things: https://github.com/ceph/ceph/blob/b7e57a079ac0ba387b8c2ec7ad93df37c02fd90d/debian/rules#L34

Maybe introduce `pkg_get_variable` in another PR?